test/crypto/CryptoDerivationTest.test.ts

```diff
@@ -0,0 +1,266 @@
+import {
+    CoreBuffer,
+    CryptoDerivation,
+    CryptoDerivationAlgorithm,
+    CryptoEncryptionAlgorithm,
+    CryptoHash,
+    CryptoHashAlgorithm,
+    CryptoSecretKey,
+    Encoding,
+    ICoreBuffer
+} from "@nmshd/crypto";
+import { expect } from "chai";
+
+export class CryptoDerivationTest {
+    public static run(): void {
+        describe("CryptoDerivation", function () {
+            describe("Execute deriveKeyFromBase()", function () {
+                let keybuffer: ICoreBuffer;
+                before(async function () {
+                    // Create 256bit entropy
+                    keybuffer = await CryptoHash.hash(
+                        CoreBuffer.fromString("test", Encoding.Utf8),
+                        CryptoHashAlgorithm.SHA256
+                    );
+                });
+
+                it("should derive the same key of same input", async function () {
+                    const derived = await CryptoDerivation.deriveKeyFromBase(keybuffer, 0, "12345678");
+                    expect(derived).to.be.instanceOf(CryptoSecretKey);
+                    expect(derived.toBase64()).to.equal(
+                        "eyJrZXkiOiJfdkg4NkdJeU12cFdnZkJ2Y09FY0dfNHRhakpiVXJ5cGRUWW9iZl9jb0Y4IiwiYWxnIjozLCJAdHlwZSI6IkNyeXB0b1NlY3JldEtleSJ9"
+                    );
+
+                    const derivedComparison = await CryptoDerivation.deriveKeyFromBase(keybuffer, 0, "12345678");
+                    expect(derivedComparison).to.be.instanceOf(CryptoSecretKey);
+                    expect(derivedComparison.toBase64()).to.equal(
+                        "eyJrZXkiOiJfdkg4NkdJeU12cFdnZkJ2Y09FY0dfNHRhakpiVXJ5cGRUWW9iZl9jb0Y4IiwiYWxnIjozLCJAdHlwZSI6IkNyeXB0b1NlY3JldEtleSJ9"
+                    );
+
+                    expect(derived.toBase64()).to.equal(derivedComparison.toBase64());
```

Comment:
Those equality checks (and later inequality checks) are redundant

Reply:
I've removed the redundant string equality checks. I left the not equal test as they are, as I would like to see if anything changes here in future versions (which shouldn't be the case). If that is the case, we know which derived value did change (and possibly why) and not only that they are not equal.